docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,141 @@
+
+IBC is an end-to-end, connection-oriented, stateful protocol for reliable, ordered, authenticated communication between modules on separate distributed ledgers. IBC implementations are expected to be co-resident with higher-level modules and protocols on the host ledger. Ledgers hosting IBC must provide a certain set of functions for consensus transcript verification and accumulator proof generation, and IBC packet relayers (off-chain processes) are expected to have access to network protocols and physical datalinks as required to read the state of one ledger and submit data to another.
+
+### Dataflow layers
+
+```
++--------------------------+                           +--------------------------+
+| Distributed Ledger A     |                           | Distributed Ledger B     |
+|                          |                           |                          |
+| +----------+     +-----+ |        +---------+        | +-----+     +----------+ |
+| | Module A | <-> | IBC | | <----> | Relayer | <----> | | IBC | <-> | Module B | |
+| +----------+     +-----+ |        +---------+        | +-----+     +----------+ |
++--------------------------+                           +--------------------------+
+```
+
+## Scope
+
+IBC handles authentication, transport, and ordering of structured data packets relayed between modules on separate ledgers. The protocol is intended to be in simultaneous use between any number of modules on any number of ledgers over arbitrarily structured underlying networks.
+
+## Interfaces
+
+IBC sits between modules — smart contracts, state machine components, or otherwise independent pieces of application logic on ledgers — on one side, and underlying consensus protocols, ledgers, and network infrastructure (e.g. TCP/IP), on the other side.
+
+To modules IBC provides a set of functions much like the functions which might be provided to a module for interacting with another module on the same ledger: sending data packets and receiving data packets on an established connection & channel — in addition to calls to manage the protocol state: opening and closing connections and channels, choosing connection, channel, and packet delivery options. Considerable flexibility is provided to ledger developers as to which of these functions to expose to which modules, and how to restrict parameter choices — if at all — the protocol generally assumes the most permissionless setting possible, and implementers can choose to restrict usage according to their application's requirements.
```

Comment:
Can you just list here operations that are provided to modules (opening/closing connections and channels(what is a channel?, and sending and receiving data)? The last sentence is maybe not needed at this point. This can be explained in part where you discuss interface in more details.

Reply:
Fair, removed the last sentence.